backend/fms_core/services/library.py

```diff
@@ -0,0 +1,113 @@
+from django.core.exceptions import ValidationError
+
+from fms_core.models import LibraryType, Library
+
+from fms_core.services.process import create_process
+from fms_core.services.sample import process_library_sample
+from fms_core.services.property_value import create_process_properties
+
+from datetime import datetime
+
+
+def get_library_type(name):
+    library_type = None
+    errors = []
+    warnings = []
+    try:
+        library_type = LibraryType.objects.get(name=name)
+    except LibraryType.DoesNotExist as e:
+        errors.append(f"No Library Type named {name} could be found.")
+    except LibraryType.MultipleObjectsReturned as e:
+        errors.append(f"More than one Library Type was found with the name {name}.")
+
+    return library_type, errors, warnings
+
+def create_library(library_type, index, platform, strandedness, library_size=None):
+    library = None
+    errors = []
+    warnings = []
+
+    if not library_type:
+        errors.append('Missing library type.')
+
+    if not index:
+        errors.append('Missing index.')
+
+    if not platform:
+        errors.append('Missing platform.')
+
+    if not strandedness:
+        errors.append('Missing strandedness.')
+
+    if errors:
+        return library, errors, warnings
+
+    try:
+        library = Library.objects.create(library_type=library_type,
+                                         library_size=library_size,
+                                         index=index,
+                                         platform=platform,
+                                         strandedness=strandedness)
+    except ValidationError as e:
+        errors.append(';'.join(e.messages))
+
+    return library, errors, warnings
+
+
+def prepare_library(library_type, library_date, platform, index, strandedness, library_volume, library_comment, container,
+                    container_coordinates, source_sample, volume_used, protocol, process_properties, process_comment=None):
+    process = None
+    errors = []
+    warnings = []
+
+    if not protocol:
+        errors.append("Missing protocol.")
+        return process, errors, warnings
+
+    if not process_properties:
+        errors.append("Missing process properties.")
+        return process, errors, warnings
+
+    process_by_protocol, process_errors, process_warnings = \
+        create_process(protocol=protocol,
```

Comment:
One process for each library is too much. You can initialize it (process_by_protocol) in the importer and pass it along instead of the protocol